### ISSUE TITLE
Error loading/downloading mod with 0.70.0

### ISSUE THREAD

**Issue by PhiMue86** [Nexus-Mods/Nexus-Mod-Manager#731]
When starting up 0.70.0, I get the error that a mod could not be loaded. Trying to add it manually or redownload it leads to the error message that the path or file name are too long. But while the file name is relatively long (67 characters), there are longer ones in my mods folder that load correctly, and the complete path including the file name is only 138 characters long. This mod loads correctly in the previous version (0.65.11).

It is the main file of this mod: https://www.nexusmods.com/morrowind/mods/45432

I have to give it a much shorter file name to be loaded correctly, much shorter than many other files that are loaded without issues.

**Comment by squid-box** [comment]
Did your previous installation point to the same location for mod files etc? Looking at the archive of the mod, the inner files end up with respectably long names:

`Enhanced Water Shader for MGE XE 2.0 Green-Blue Water\Enhanced Water Shader for MGE XE 2.0 Green-Blue Water without Sewer Waves\Data files\shaders\XEshaders\Underwater Effects.fx` 
is 178 characters long.

Could you [force a trace log to be created](https://github.com/Nexus-Mods/Nexus-Mod-Manager/wiki/Bug-reporting#trace-log) and attach it here, hopefully it would point to which file is having a problem.

**Comment by PhiMue86** [comment]
I just realized that I used a different folder with shorter Path in the older version - that means, both are using the same folder, but I created a junction with a shorter path to solve this exact issue, because I didn't want to move the folder itself to this location. When I point the new NMM version to the same junction, it loads the file correctly. Sorry about the false alarm... I was just puzzled by the fact that other files with longer names were working fine. However, I didn't think about the folder path inside the archive.

By the way, I don't have any mods installed with NMM, I only use it for tracing mod updates.

I am still having problems with loading a number of Oblivion mods, though (not a big issue as I don't play Oblivion at the moment and when I did I was using Mod Organizer), and that doesn't seem to have anything to do with the length of the file names or paths. I'll try to find out anything about that.

I also just tried searching for updates of my Morrowind mods, but it stopped after a short time with an error. Unfortunately, it's in German, not sure if that would be of any use to you. It says "Die Eingabezeichenfolge hat das falsche Format", which means "input string was not in a correct format". There are more German error messages in the TraceLog. I'm going to try to find out which mod is causing it to crash. Searching for updates works in the older version.

**Comment by squid-box** [comment]
No worries, it's good to get detailed error reports.

There's been changes in the update searches due to the new API, please create a new issue here for it and attach the trace log (I'll work around the German) :)

**Comment by PhiMue86** [comment]
I've been trying to narrow it down but now it says API Rate Limit exceeded when I try checking for updates. I've waited for an hour but the message reappeared almost immediately. How long do I have to wait before I can make new checks?

**Comment by squid-box** [comment]
There's a little info icon at the bottom which will show the details, but you get 2500 calls per day (resets at midnight UTC) and once that limit is hit you get 100 calls per hour (resets on the hour).

For my knowledge: how many mods do you have added?

We will optimize the number of calls made, but had to release yesterday to not get caught with nothing working for when the old API was shut down.

**Comment by PhiMue86** [comment]
> For my knowledge: how many mods do you have added?

A lot. XD However, I've disabled update checking for many files, because they are not from Nexus or because they are very old and extremely unlikely to be updated, or because they are optional files which don't follow the main mod's versioning scheme. But in order to find the file responsible for the error I temporarily enabled it for all files, of which there are about 850 at the moment (I've intended to remove unneeded files for some time, but unfortunately they only get more...), so yes, I can see how this can be a problem. I wasn't aware of the limit. Under normal circumstances it would be enough, I only check for updates every few days. And now that I know about the limit I can try to disable update checking for more files. It takes too long to check for all of them, anyway. ;)

So, now I'm going to disable the checks for all files and only enable some at a time to find out which causes the problem.

**Comment by squid-box** [comment]
Everyone has their own use-case, I hover around 50 mods, and you're at 850. All are equally valid. I created #734 to help with your use-case, hopefully we can address if fairly soon.

**Comment by PhiMue86** [comment]
It would be better if I only checked updates for mods I actually use, but I'm not even playing the game at the moment, only modding it. :D Actually, the main reason I am using NMM is just because it checks all mods and not only the installed ones.

**Comment by PhiMue86** [comment]
Ok, now I have disabled update checking for all but a very small number of mods, but checking for updates seems to remove the info for these files, the version numbers become grey and no longer link to the mod's pages and they are no longer correctly named. All files of a mod are just set to the mod's overall name, but the actual file's name is missing. "Get missing mod info" doesn't work any more.

On the plus side, I may have found the files that were causing the issue, although I have no idea why. It's this very old mod: https://www.nexusmods.com/morrowind/mods/9967

Checking updates leads to the previously mentioned error, the same happens when I click "Get missing mod info" or when I try to download the files again.

Edit: Here's the trace log for the error: [TraceLog20190513174132.txt](https://github.com/Nexus-Mods/Nexus-Mod-Manager/files/3173706/TraceLog20190513174132.txt)

**Comment by squid-box** [comment]
The error in the log is due to NMM not parsing the version number "1.0." correctly, that's definitely a bug (and these non-standard version numbers are killing me).

**Comment by PhiMue86** [comment]
Ah, good, then I'm glad I could help. Yeah, some mods have pretty unconventional version numbers. :D

**Comment by squid-box** [comment]
@Pherim: Could you please verify if your use-cases have been resolved with the latest version, or otherwise provide me with a new tracelog?

**Comment by PhiMue86** [comment]
Yes, downloading the mod with the version number ending in a "." works, and the other issue was my own fault because I used the wrong folder path. ;D

Anyway, I am still having problems with newly downloaded mods not being linked to their nexus pages and it seems checking for updates doesn't work at all. But I already posted this in a different thread.

**Comment by squid-box** [comment]
I've submitted a fix for the links not working (#780), as for the updates not being found - I'd love to know which mod(s) you're seeing this with - it would make it much easier to troubleshoot.

**Comment by PhiMue86** [comment]
Looks like it doesn't work at all. Two mods with updates that are not found are these, for example:

https://www.nexusmods.com/morrowind/mods/45952
https://www.nexusmods.com/morrowind/mods/45969

But I only know that because I went looking for mods that had been updated since my last download. There are probably more. As I said it doesn't find any updates at all.

"Find missing mod info" also doesn't work, for  mods that are added manually because they don't have the "Mod Manager Download" button, like the second mod I posted above.

**Comment by squid-box** [comment]
Sorry, I wasn't very clear, the fix is in the pipeline to be included in the next version - 0.70.6

Edit: And it's too early for me, thanks for the mods, I'll investigate further tonight.

**Comment by PhiMue86** [comment]
Downloaded mods are now linked correctly in 0.70.6, and "Find missing mod info" also works for previously downloaded mods that did not have links. But it doesn't work for the second mod I posted above (Tooltip) which I had to download and add manually. It used work in previous versions with the same mod.

**Comment by PhiMue86** [comment]
So, any news about the missing mod info problem for the tooltip mod? Why are there no matching mods displayed?
